rcon/automods/seeding_rules.py

```diff
@@ -257,16 +259,40 @@ def punitions_to_apply(
                 if server_player_count >= dwc.max_players:
                     self._disable_for_round("disallowed_weapons")
 
+                ecf = self.config.enforce_cap_fight
+                if server_player_count >= ecf.max_players:
+                    self._disable_for_round("enforce_cap_fight")
+
                 violations = []
                 if (
-                    not self._is_seeding_rule_disabled("disallowed_roles")
-                    and drc.min_players <= server_player_count < drc.max_players
+                        not self._is_seeding_rule_disabled("disallowed_roles")
+                        and drc.min_players <= server_player_count < drc.max_players
                 ):
                     if aplayer.role in drc.roles:
                         violations.append(
                             drc.message.format(role=drc.roles.get(aplayer.role))
                         )
 
+                if not self._is_seeding_rule_disabled("enforce_cap_fight") and \
+                        (team == "axis" and game_state["axis_score"]) >= ecf.max_caps or \
+                        (team == "allies" and game_state["allied_score"] >= ecf.max_caps):
+                    self.logger.debug("Player is on " + team + " side and winning")
+                    op = player['offense']
+                    oop = watch_status.offensive_points.setdefault(aplayer.name, 0)
+
+                    self.logger.debug(
+                        "Player had " + str(oop) + " offensive points last and now " + str(op))
+                    if oop != 0 and oop < op:
```

Comment:
Correct me if I'm wrong, but whenever this rule becomes active you're not populating the watch status with the current offense score of players and instead default it to 0. If any players on the server already have an offense score before the seeding automod is enabled, either because the min/max player requirements are met or the service was just started, it may falsely flag them as attacking and warn or punish them.

Reply:
I think that is kind of right.

The seeding automod cannot go into "active" once it was deactivated once for the round (that prevents situations, especially with other rules, that a tank crew is punished because they used a tank when the player requirement allowed it and then it drops below the requirement).
However, when the automod is started _during_ a round which matches the seeding requirements, I think you're right, that it will punish players with a pre-existing Offensive point. Let me take a look how to work that in, so that this does not happen. I think changing the default value of `oop` (old offensive points) from 0 (which is a valid in-game value) to -1 might be the easiest solution (while ignoring an oop of -1 in the "should punish check").